lib/galaxy/tools/__init__.py

```diff
@@ -1766,12 +1766,19 @@ def expand_incoming(self, trans, incoming, request_context, input_format="legacy
 
         # Remapping a single job to many jobs doesn't make sense, so disable
         # remap if multi-runs of tools are being used.
-        if rerun_remap_job_id and len(expanded_incomings) > 1:
-            raise exceptions.MessageException(
+        produces_multiple_jobs = len(expanded_incomings) > 1
+        if rerun_remap_job_id and produces_multiple_jobs:
+            raise exceptions.RequestParameterInvalidException(
                 "Failure executing tool with id '%s' (cannot create multiple jobs when remapping existing job).",
                 self.id,
```

Comment:
This looks wrong, it probably was a log statement before.

```suggestion
                f"Failure executing tool with id '{self.id}' (cannot create multiple jobs when remapping existing job)."
```